Sources/MapboxNavigation/RouteLineController.swift

```diff
@@ -21,8 +21,6 @@ extension NavigationMapView {
             navigationViewData.router
         }
         
-        fileprivate var routeProgress: RouteProgress?
```

Comment:
instead of storing the routeProgress to update vanishing route line, when the congestion level/route/current leg index is not changed, we only use the location coordinate to truncate the existing gradient stops.